src/rgw/rgw_tracer.cc

```diff
@@ -0,0 +1,82 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab ft=cpp
+
+#include "rgw_tracer.h"
+#include <yaml-cpp/yaml.h>
+
+thread_local std::unique_ptr<tracing::Tracer> rgw_tracer = std::unique_ptr<tracing::Tracer>(new tracing::Tracer(jaeger_configuration::jaeger_default_config));
+
+namespace tracing {
+
+const std::shared_ptr<opentracing::Tracer> Tracer::noop_tracer = opentracing::MakeNoopTracer();
+TracerConfiguration tracing::Tracer::tracer_conf;
+
+Tracer::Tracer(jaegertracing::Config& conf):open_tracer(jaegertracing::Tracer::make(conf)) {};
+
+Tracer::Tracer(const std::string& config_file_path) {
+  const auto config_yaml = YAML::LoadFile(config_file_path);
+  const auto config = jaegertracing::Config::parse(config_yaml);
+  open_tracer =  jaegertracing::Tracer::make(config);
+};
+
+Tracer::~Tracer() {}
```

Comment:
no need to implement or declare a default dtor.

Reply:
sure. I forgot to delete it.